remotecommanddispatcher/src/main/java/com/tealium/remotecommanddispatcher/RemoteCommandParser.kt

```diff
@@ -12,33 +12,31 @@ class RemoteCommandParser {
          */
         fun mapPayload(payload: Map<String, Any>, lookup: Map<String, String>): MutableMap<String, Any> {
             val mappedPayload = mutableMapOf<String, Any>()
-            lookup.forEach { (key, value) ->
-                payload[key]?.let { payloadValue ->
-                    lookup[key]?.let { lookupValue ->
-                        checkAndSplitDestinationList(lookupValue).forEach { valuesList ->
-                            val objectRow = splitKeys(valuesList, payloadValue, lookup)
-                            val objectKey = objectRow.second
-                            objectKey?.let {
-                                if (mappedPayload.containsKey(it)) {
-                                    // object key is already in the map, append to the same key
-                                    (mappedPayload[it] as? MutableMap<*, *>)?.let { objectMap ->
-                                        // create a map with a String key. This will throw an exception if the JSON mapping file does not use a String as a key.
-                                        val oMap = objectMap.entries.associate { entry -> entry.key as String to entry.value }.toMutableMap()
-                                        (objectRow.first[objectKey] as? Map<*, *>)?.let {
-                                            it.entries.associate { entry -> entry.key as String to entry.value } // as String }
-                                        }?.forEach { (kk, vv) ->
-                                            // add mapped values from splitKeys to the temporary oMap
-                                            oMap[kk] = vv
-                                            // append to the same key (e.g. "event")
-                                            mappedPayload[objectKey] = oMap
-                                        }
+            lookup.forEach { (lookupKey, lookupDestination) ->
+                payload[lookupKey]?.let { payloadValue ->
+                    checkAndSplitDestinationList(lookupDestination).forEach { destinations ->
+                        val objectRow = splitKeys(destinations, payloadValue)
+                        val objectKey = objectRow.second
+                        objectKey?.let {
+                            if (mappedPayload.containsKey(it)) {
+                                // object key is already in the map, append to the same key
+                                (mappedPayload[it] as? MutableMap<*, *>)?.let { objectMap ->
+                                    // create a map with a String key. This will throw an exception if the JSON mapping file does not use a String as a key.
+                                    val oMap = objectMap.entries.associate { entry -> entry.key.toString() to entry.value }.toMutableMap()
+                                    (objectRow.first[it] as? Map<*, *>)?.let {
+                                        it.entries.associate { entry -> entry.key.toString() to entry.value } // as String }
```

Comment:
Is the `// as String }` bit at the end needed?

Reply:
that is actually from the main feature branch - maybe worth enabling the `ignore whitespace changes` feature, as the default git diff suggests I've made more changes than I actually have. I've removed one of the closures because it was fetching the `destination` from the `lookup` but that had already been retrieved in the outermost `forEach`

have now removed that bit of code though.